ako-gateway-api/nodes/gateway_model.go

```diff
@@ -174,19 +179,28 @@ func BuildVsVipNodeForGateway(gateway *gatewayv1.Gateway, vsName string) *nodes.
 	return vsvipNode
 }
 
-func DeleteTLSNode(key string, object *AviObjectGraph, gateway *gatewayv1.Gateway, secretObj *corev1.Secret, encodedCertNameIndexMap map[string][]int) {
+func DeleteTLSNode(key string, object *AviObjectGraph, gateway *gatewayv1.Gateway, secretObj *corev1.Secret, encodedCertNameIndexMap map[string][]int) bool {
 	var tlsNodes []*nodes.AviTLSKeyCertNode
 	_, _, secretName := lib.ExtractTypeNameNamespace(key)
 	evhVsCertRefs := object.GetAviEvhVS()[0].SSLKeyCertRefs
 	gwStatus := akogatewayapiobjects.GatewayApiLister().GetGatewayToGatewayStatusMapping(gateway.Namespace + "/" + gateway.Name)
+	totalListeners := len(gateway.Spec.Listeners)
+	invalidListeners := 0
 	for i, listener := range gateway.Spec.Listeners {
-		if akogatewayapilib.IsListenerInvalid(gwStatus, i) {
+		// Check Listener conditions validity for non-tls type
+		if listener.TLS == nil && akogatewayapilib.IsListenerInvalid(gwStatus, i) {
```

Comment:
since the IsListenerInvalid check also being done in TLS != nil case, we can keep the old condition and move it to line 219 and remove condition on line 216

Reply:
done